pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -391,10 +211,20 @@ def _build_operator(
                 else:
                     operator = qutip.Qobj(operator).to("CSR")
                 for qubit in qubits:
-                    k = self._qid_index[qubit]
+                    k = self.data._qid_index[qubit]
                     op_list[k] = operator
         return qutip.tensor(op_list)
 
+    @property
+    def basis_name(self) -> str:
+        """What states are used in the Hamiltonian."""
+        return self.data.basis_name
```

Comment:
This doesn't make much sense, or at the very least it doesn't match the docstring. Can we get rid of it perhaps?